File: go/0242E-有效的字母异位词.go
/*
 * @Author: SukiEva
 * @Date: 2021-10-16 15:21:06
 * @Description: https://github.com/SukiEva
 * わたし、気になります！
 */

package main

func isAnagram(s string, t string) bool {
	if len(s) != len(t) {
		return false
	}
	cnt := make(map[byte]int)
	for _, val := range s {
		cnt[byte(val)]++
	}
	for _, val := range t {
		cnt[byte(val)]--
		if cnt[byte(val)] < 0 {
			return false
		}
	}
	return true
}

File: cpp/0242E-有效的字母异位词.cpp
/*
 * @Author: SukiEva
 * @Date: 2021-12-14 08:43:36
 * @Description: https://github.com/SukiEva
 * わたし、気になります！
 */
#include <bits/stdc++.h>

#include "include/ListNode.h"
#include "include/TreeNode.h"

using namespace std;

class Solution {
   public:
    bool isAnagram(string s, string t) {
        if (s.size() != t.size()) {
            return false;
        }
        map<char, int> mp;
        for (auto c : s) mp[c]++;

        for (auto c : t) {
            mp[c]--;
            if (mp[c] < 0) return false;
        }
        return true;
    }
};